flake.nix

```diff
@@ -0,0 +1,42 @@
+{
+  inputs = {
+    utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs, utils }:
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+      in
+      {
+        devShells.default = pkgs.mkShell {
+          packages = with pkgs; [
+            python311Full
+            python311Packages.pip
+            python311Packages.magic
+            python311Packages.virtualenv
+
+            nodePackages.npm
+            nodejs_21
+
+            gettext
+            netcat-gnu
+            pcre16
+            file
```

Comment:
Hey @charludo, this looks like a very beneficial addition. As someone currently undergoing the onboarding process, I recently set up my project and encountered a few hurdles, two of which would be addressed by this pull request. Let me outline the issues I faced, which I believe might also affect future macOS users:

- I initially encountered compatibility issues due to a newer Python version, which took some time to resolve. Fortunately, this pull request resolves this issue by pinning the Python version.
- On macOS, the default sed version differs from the one expected by translate.th, leading to potential issues. Remedying this typically involves manually installing GNU sed using brew install gnu-sed, or alternatively, incorporating it into this script as suggested. This [asciinema](https://asciinema.org/a/0mDY357AXvo5s55C7vhBauqEo) demonstrates that the issue is successfully resolved upon executing nix develop.
- Additionally, my macOS setup lacked the libmagic library, which is utilized by certain components of the system. Unfortunately, there isn't a Nix package readily available to provide this library, so this particular issue remains unresolved for now.

```suggestion
            file
            gnused
```

In summary integrating gnu-sed into the script would significantly alleviate macOS-related compatibility issues encountered by users. Moreover, if we were to develop and publish a Nix package for libmagic and incorporate it into this pull request, we could effectively resolve all remaining macOS-related hurdles. 😌

Reply:
Hi @deen13, thanks for the feedback, very glad to hear this is already useful!! 😄

Funnily enough, I did also encounter libmagic-related issues when attempting to get the project running under NixOS, but this flake did fix those issues for me.

I will have another look at that specifically. Did you resolve your issues by just installing a MacOS version of libmagic globally? Or was there another workaround?

Reply:
@deen13 unfortunately I don't have MacOS available to test this.

Would you be OK checking if adding
```nix
python311Packages.magic
python311Packages.python-magic
python311Packages.typecode-libmagic
python311Packages.typecode
```

To the packages list resolves the issue? 🙈

Reply:
Hey @charludo, sure I'll check if it works once I'm at my computer later this day. I resolved this issues, as you already said, by installing libmagic globally via brew.

Reply:
Hey @charludo, [nix develop](url) is failing on my system with an error, which you can see in the attached screenshot. It appears that it's attempting to use a native version of libmagic. What are your thoughts?

![image](https://github.com/digitalfabrik/integreat-cms/assets/13850172/3f92e9ee-e05d-421a-9e8d-8fd55964ee9c)

Reply:
Hm, looks like `python311Packages.typecode-libmagic` is just [not packaged for MacOS](https://github.com/NixOS/nixpkgs/blob/1487bdea619e4a7a53a4590c475deabb5a9d1bfb/pkgs/development/python-modules/typecode/libmagic.nix#L48) :(

I don't think we should attempt allowing unsupported systems's inputs, so unfortunately my only idea is to check if omitting that package does the job still, and same thing if the build fails again due to `python311Packages.typecode` (which it might, apparently it pulls in `python311Packages.typecode-libmagic` anyways).

Maybe `python311Packages.python-magic` is sufficient to solve the problem? 

Sorry for being so vague, without a system to test this on I'm tapping in the dark :(

Reply:
I've already tried `python311Packages.python-magic` and this does not work neither. I totally understand the problems you're facing w/o having a mac to test on. I'm currently playing with devcontainers in order to solve the issues specifically for mac users. Maybe in the future nix flakes and devcontainers can co-exist happily. 🤗

Reply:
Yesss that's all. ☺️

Reply:
Sorry to bother you again @deen13, but would you mind trying the flake again? I was able to replicate the missing `libmagic` by entering the shell in "pure" mode (`nix-develop --unset PATH`), and have been able to fix the issue (by passing `file` into the `LD_LIBRARY_PATH`). 

This + the new automatic postgres setup *should* solve the remaining issues.

Reply:
Hey @charludo, I pulled the latest changes and ran `nix develop && ./tools/run.sh` again but it still fails to find libmagic. :/